jcsv.go

```diff
@@ -1,11 +1,83 @@
 package jcsv
 
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+)
+
 func JsonToCsv(j []byte, addHeaders bool) ([]byte, error) {
-	// TODO: convert JSON data in `j` into CSV format and return
-	return nil, nil
+	// create file object
+	var JSON file
+
+	// convert data in j to Data
+	err := json.Unmarshal(j, &JSON.data)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Again, it will have issues like the input is an object or an array.
Why don't you add two more functions for `file`.
1. `ParseCsv(c []byte, hasHeaders bool) (file, error)`, call it inside `ParseOpenedCsvFile()` and here.
2. `ParseJson(j []byte) (file, error)`, call it inside `ParseOpenedJsonFile()` and inside `CsvToJson()` below.

Reply:
I created the function ParseJson. I called it in ParseOpenedJsonFile and JsonToCsv.  It resolved the issue you identified in JsonToCsv.